workspace/notebook/rebuild_tables.json

```diff
@@ -0,0 +1,903 @@
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"standardised_container = 'odw-standardised'\r\n",
+					"harmonised_container = 'odw-harmonised'\r\n",
+					"curated_container = 'odw-curated'\r\n",
+					"curated_migration_container = 'odw-curated-migration'\r\n",
+					"config_container = 'odw-config-db'\r\n",
+					"logging_container = 'logging'"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "code",
+				"metadata": {
+					"jupyter": {
+						"source_hidden": false,
+						"outputs_hidden": false
+					},
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"def rebuild_table(db_name: str, table_name: str, location: str, table_format: str):\r\n",
+					"    spark.sql(f\"\"\"CREATE TABLE IF NOT EXISTS {db_name}.{table_name}\r\n",
+					"    USING {table_format}\r\n",
+					"    LOCATION '{location}'\r\n",
+					"    \"\"\")\r\n",
+					"    print(f\"Table {db_name}.{table_name} rebuilt at location\\n{location}\")"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "code",
+				"metadata": {
+					"jupyter": {
+						"source_hidden": false,
+						"outputs_hidden": false
+					},
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"def list_tables(db_name: str):\r\n",
+					"    spark.sql(f\"USE {db_name}\")\r\n",
+					"    tables_df: DataFrame = spark.sql(\"SHOW TABLES\")\r\n",
+					"    table_names: list = [row['tableName'] for row in tables_df.collect()]\r\n",
+					"    \r\n",
+					"    return table_names"
+				],
+				"execution_count": null
+			},
+			{
+				"cell_type": "code",
+				"metadata": {
+					"jupyter": {
+						"source_hidden": false,
+						"outputs_hidden": false
+					},
+					"nteract": {
+						"transient": {
+							"deleting": false
+						}
+					}
+				},
+				"source": [
+					"def create_table_metadata(container: str, db_name: str, tables: list) -> dict:\r\n",
+					"    tables = list_tables(db_name)\r\n",
+					"    table_metadata = {}\r\n",
+					"    for table in tables:\r\n",
+					"        try:\r\n",
+					"            table_details = spark.sql(f\"DESCRIBE DETAIL {db_name}.{table}\")\r\n",
+					"            table_info = table_details.collect()[0].asDict()\r\n",
+					"            table_name = table_info.get(\"name\", table)\r\n",
+					"            table_location = table_info.get(\"location\", \"Unknown\")\r\n",
+					"            # This handles cases where tables were created in the default location by mistake by not specifying location in the create table script. \r\n",
+					"            # Therefore there were tables created in the default location. This won't be needed in future.\r\n",
+					"            if table_location.startswith('abfss://synapse'):\r\n",
+					"                location = f'abfss://{container}@pinsstodwdevuks9h80mb.dfs.core.windows.net/{table_name}'\r\n",
```

Comment:
the storage account name could be parameterised too using `py_utils_get_storage_account` notebook if we aim to use this notebook in other envs.

Reply:
Done